conda/recipes/cudf_kafka/meta.yaml

```diff
@@ -60,6 +60,13 @@ requirements:
     - cuda-version ={{ cuda_version }}
     - cudf ={{ version }}
     - libcudf_kafka ={{ version }}
+    # Needed by rapids-build-backend to infer the CTK version
+    {% if cuda_major == "11" %}
+    - cudatoolkit
+    {% else %}
+    - cuda-nvcc-impl
```

Comment:
I don't actually know the difference between those packages. This block was already here in the PR when I picked it up and I didn't question it.

Could you help me understand the difference and why `cuda-nvcc` might be preferred?

I did just read through https://github.com/conda-forge/cuda-nvcc-feedstock/issues/12 and https://github.com/conda-forge/cuda-nvcc-feedstock/pull/13 and don't really understand the different purposes for these packages.

Reply:
> This block was already here in the PR when I picked it up and I didn't question it.

Ah. @vyasr Can you explain why you chose `cuda-nvcc-impl` over `cuda-nvcc`? Is this something we should put into the (draft) documentation for CUDA conda packages?

Reply:
We briefly discussed this in a meeting -- if this is _only_ to provide `nvcc` for version checks, we can probably simplify. For CUDA 11, `cudatoolkit` doesn't actually ship `nvcc`. That has to come from the system, so let's remove the `cudatoolkit` dependency. For CUDA 12, we could switch to `cuda-nvcc` or leave this as-is.

Longer-term question: Could we make `rapids-build-backend` aware of the `RAPIDS_CUDA_VERSION` environment variable, or provide some other means of specifying the CUDA version? It's a bit awkward that we have to install compiler packages just to indicate the desired CUDA version.

Reply:
> *It's a bit awkward that we have to install compiler packages just to indicate the desired CUDA version.*

I realized we could simplify **even further**... conda builds don't need `nvcc` available at all for `rapids-build-backend` purposes, because in `build.sh` we're doing this:

```shell
python -m pip install \
    --no-build-isolation \
    --no-deps \
    --config-settings rapidsai.disable-cuda=true
```

And that `rapidsai.disable-cuda=true` skips the code path in `rapids-build-backend` that needs `nvcc`.

https://github.com/rapidsai/rapids-build-backend/blob/c6ef8cf988d6572e7a881c4bcd4fc376ba3b50df/rapids_build_backend/impls.py#L154-L155

Relevant PR where that `disable-cuda` option was added: https://github.com/rapidsai/rapids-build-backend/pull/32.

Double-checked the 2 already-merged PRs... neither of them added new dependencies for conda builds, other than `rapids-build-backend` itself.

* https://github.com/rapidsai/raft/pull/2331
* https://github.com/rapidsai/rmm/pull/1502

So I think this is resolved, and `rapids-build-backend` doesn't need any changes to account for it.